### PR TITLE
Update subgraph with delegation queries per account and per pool

### PR DESCRIPTION
**Pull request by Robert-H-Leonard**

### CODE REVIEW

File: subgraph/src/datasources/FortaStaking.ts [hunk 1]
```diff
@@ -101,9 +143,17 @@ function updateStake(
...
+  if (account == null) {
+    account = new Account(_stakerId);
+  }
+
   if (stake == null) {
     stake = new Stake(_subjectId + _stakerId);
   }
+
+  const prevStakeTotalShares = stake.shares ? (stake.shares as BigInt).toI32() : 0;
+  const prevStateInActiveShares = stake.inactiveShares ? (stake.inactiveShares as BigInt).toI32(): 0;
```
Comment (Robert-H-Leonard) [inline subgraph/src/datasources/FortaStaking.ts:155]: Are total shares and active shares the same as total stake and active stake?

File: subgraph/src/schema.gql [hunk 2]
```diff
@@ -163,6 +164,7 @@ type Subject @entity {
   slashedTotal: Int
   stakes: [Stake!]! @derivedFrom(field: "subject")
   stakeDepositedEvents: [StakeDepositEvent!] @derivedFrom(field: "subject")
+  withdrawalExecutedEvent: [WithdrawalExecutedEvent!] @derivedFrom(field: "subject")
```
Comment (ivandiazwm) [inline subgraph/src/schema.gql:167]: name should be plural `withdrawalExecutedEvents`

File: subgraph/src/datasources/FortaStaking.ts [hunk 3]
```diff
@@ -147,13 +208,20 @@ export function handleWithdrawalInitiated(event: WithdrawalInitiated): void {
...
-  updateStake(
+export function handleWithdrawalExecuted(event: WithdrawalExecuted): void {  
+  const stakeId = updateStake(
     event.address,
     event.params.subjectType,
     event.params.subject,
     event.params.account,
   );
+
+  const withdrawlExecutedEvent = new WithdrawalExecutedEvent(events.id(event));
```
Comment (ivandiazwm) [inline subgraph/src/datasources/FortaStaking.ts:219]: minor type here too `withdrawalExecutedEvent`

File: subgraph/src/datasources/FortaStaking.ts [hunk 4]
```diff
@@ -117,24 +166,36 @@ function updateStake(
     _subject,
     _staker
   );
+
+  const updatedStakeTotalShares: BigInt = stake.shares ? stake.shares as BigInt : BigInt.fromI32(0);
+  const updatedStateInActiveShares: BigInt = stake.inactiveShares ? stake.inactiveShares as BigInt: BigInt.fromI32(0);
+
+  updateAggregateStake(_stakerId, prevStakeTotalShares, prevStateInActiveShares, updatedStakeTotalShares, updatedStateInActiveShares)
```
Comment (ivandiazwm) [inline subgraph/src/datasources/FortaStaking.ts:173]: so, we should use it in the way for scanner pools, correct?

**Review by ivandiazwm — APPROVED** [review]

File: subgraph/src/datasources/FortaStaking.ts (hunk 4 above)
Comment (Robert-H-Leonard) [reply subgraph/src/datasources/FortaStaking.ts:173]: Yep I think we could reuse the same function.

File: subgraph/src/datasources/FortaStaking.ts (hunk 3 above)
Comment (Robert-H-Leonard) [reply subgraph/src/datasources/FortaStaking.ts]: Updated

**Pull request merged**